overlord/snapstate/check_snap.go

```diff
@@ -433,6 +445,34 @@ func earlyEpochCheck(info *snap.Info, snapst *SnapState) error {
 	return checkEpochs(nil, info, cur, Flags{}, nil)
 }
 
+// check that the listed system users are valid
+var (
+	findUid = osutil.FindUid
+	findGid = osutil.FindGid
+)
+
+func checkSystemUsernames(si *snap.Info) error {
+	for _, user := range si.SystemUsernames {
+		if !supportedSystemUsernames[user.Name] {
+			return fmt.Errorf(`Unsupported system username "%s"`, user.Name)
```

Comment:
go error messages should always start with lowercase (for nesting)!! I didn't spot this, fixing this myself